questions/javascript-questions.md

```diff
@@ -64,5 +64,5 @@ duplicate([1,2,3,4,5]); // [1,2,3,4,5,1,2,3,4,5]
 * What are the benefits of using `spread syntax` and how is it different from `rest syntax`?
 * How can you share code between files?
 * Why you might want to create static class members?
-* How to implement performance improvement of a search functionality, by executing search logic only when user finished typing?
-* How to restrict a function to only execute with a gap of atleast t seconds?
+* How to implement performance improvement of a search functionality, by executing search logic only when user finished typing? (debouncing)
+* How to restrict a function to only execute with a gap of atleast t seconds? (throttling)
```

Comment:
This doesn't actually test whether or not the user understands the concept. This is nuts and bolts, not conceptual. Someone could write this code and not know what throttling is or know that there are utilities available to provide this functionality. 

Please read through the questions in this file and pattern this section after those.